sss/add_puma.py

```diff
@@ -0,0 +1,174 @@
+import os
+import glob
+import pandas as pd
+import warnings
+from pandas.core.common import SettingWithCopyWarning
+warnings.simplefilter(action="ignore", category=SettingWithCopyWarning)
+
+# declare PUMA data columns and data type
+class PUMA(Base):
+    __tablename__ = 'puma'
+    sl = Column('sl', String)
+    state_fips = Column('state_fips', String)
+    state = Column('state', String, primary_key = True)
+    puma_code = Column('puma_code', String, primary_key = True)
+    county_fips = Column('county_fips', String)
+    county_sub_fips =  Column('county_sub_fips', String)
+    county_sub = Column('county_sub', String)
+    county = Column('county', String)
+    puma_area = Column('puma_area', String)
+    place = Column('place', String, primary_key = True)
+    population_self = Column('population', Integer)
+    weight = Column('weight', Float)
+    year = Column('year', Float)
+
+
+def read_puma(path):
+    """
+    Reads puma data into data frame. The puma data stores by states in txt format.
+    
+    Parameters
+    ----------
+    path: str
+        path name of txt puma file
+    Returns
+    -------
+    pandas.datafranme
+        the returned dataframe has different levels of fips, population size etc. from txt file
+    """
+
+    # Please refer 2010_PUMA_Equivalency_Format_Layout to check he record layouts for each field 
+    colspecs = [(0,3),(3,5),(5,13),(13,18),(18,21),(21,29),(29,34),(34,42),(42,47),(47,55),(55,61),(61,70),(70,79),(79,178)]
+    names = ['sl','state_fips','state_national_standards','puma_code','county_fips','county_national_standards',
+        'county_sub_fips','county_sub_national_standards','place_fips','place_national_standards','tract_code',
+        'population','house_number','area_name']
+    # Read a table of fixed-width formatted lines from .txt into DataFrame.
+    df = pd.read_fwf(path,colspecs=colspecs,names=names,dtype=str,encoding='latin1')
+    # change data type of some variables
+    df['population'] = df['population'].astype('float')
+    df['house_number'] = df['house_number'].astype('float')
+    # add a column called year
+    df['year'] = 2010
```

Comment:
Should this be passed in rather than being hard-coded?

Reply:
Add a year parament. The year data is not in the puma.txt